winch/codegen/src/trampoline.rs

```diff
@@ -718,16 +715,15 @@ where
         // Free the stack space allocated by pushing the trampoline arguments.
         self.masm.free_stack(arg_size);
         // Restore the callee-saved registers.
-        for (r, s) in self.callee_saved_regs.iter().rev() {
-            self.masm.pop(*r, *s);
-        }
+        self.masm.restore_clobbers(&self.callee_saved_regs);
         self.masm.epilogue(0);
     }
 
     /// The trampoline's epilogue.
     fn epilogue(&mut self, arg_size: u32) {
         // Free the stack space allocated by pushing the trampoline arguments.
         self.masm.free_stack(arg_size);
+        self.masm.restore_clobbers(&[]);
```

Comment:
Should we have something like:

```suggestion
        debug_assert!(self.callee_saved_regs.is_empty());
        self.masm.restore_clobbers(&[]);
```

Reply:
That's a great idea, thanks Andrew!